scripts/west_commands/runners/stm32cubeprogrammer.py

```diff
@@ -271,15 +275,20 @@ def flash(self, **kwargs) -> None:
         if self._erase:
             self.check_call(cmd + ["--erase", "all"])
 
-        # flash image and run application
+        # Define binary to be loaded
         if self._use_elf:
+            # Use elf file if instructed to do so.
             dl_file = self.cfg.elf_file
-        elif self.cfg.bin_file is not None and os.path.isfile(self.cfg.bin_file) and \
-            "zephyr.signed" in self.cfg.bin_file:
+        elif self.cfg.bin_file is not None and self._download_address is not None:
+            # Use bin file if available and --download-address provided
```

Comment:
```suggestion
            # Use bin file if a signed binary is available
            # --download-address must be provided
```
If `.bin` is provided but not `--download-address`, maybe a warning should be logged too?